arm-asazure/2016-05-06/swagger/asazure.json

```diff
@@ -0,0 +1,562 @@
+        }
+      }
+    },
+    "RpSkuTier": {
+      "description": "Represents the SKU name and tier",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "enum": [
+            "S1",
+            "S2",
+            "S4"
+          ],
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": false
+          },
+          "description": "Name of the Sku level for the server being provisioned."
+        },
+        "tier": {
+          "type": "string",
+          "enum": [
+            "Developer",
+            "Standard"
+          ],
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": false
+          },
+          "description": "Name of the tier to which the Sku applies."
+        }
+      },
+      "required": [
+        "name",
+        "tier"
+      ]
+    },
+    "RpPropertyBag": {
+      "description": "Property bag",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The current state of the provisioning operation"
+        },
+        "serverFullName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Full URI of the Analysis Server"
+        },
+        "asAdministrators": {
+          "$ref": "#/definitions/RpAsAdministrators",
+          "description": "List of users to be made administrators on the server"
+        }
+      }
+    },
+    "RpAsAdministrators": {
+      "description": "Array of administrator user identities",
+      "type": "object",
+      "properties": {
+        "members": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "description": "UPN of the user. Example johnsmith@contoso.com."
+          },
+          "description": "Collection of administrator user identities"
+        }
+      }
+    },
+    "RpProvisionRequest": {
+      "description": "Provision request specification",
+      "type": "object",
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "Location of the Analysis server"
+        },
+        "sku": {
+          "$ref": "#/definitions/RpSkuTier",
+          "description": "Sku of the Analysis server"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": { "type": "string" },
+          "description": "Key value pairs of additional properties that can ebe specified"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RpPropertyBag",
+          "description": "Provision request properties"
+        }
+      },
+      "required": [
+        "location", 
+        "sku"
+        ]
+    },
+    "RpProvisionResult": {
+      "description": "Provision result specification",
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
..which leads me to my next observation. There is no reason to (in this case) model the PUT body parameters as a separate type from the GET body parameters - or for the response body for PUT/PATCH/GET for that matter. 

The type names in the swagger document are very RPC (verb/operation result) like. You should be modeling a (mostly :)) RESTful API where the resource type is (in this case) an AnalysisServer. The verbs are Create (or CreateOrUpdate for services that do not support PATCH), Get, List, Delete and Update (for services that support PATCH). 

I would strongly recommend that you rename the result object to correspond to the resource that they represent. The next thing is to see how far you can get by using the same model type for your PUT and GET, applying the correct readOnly:ness to server generated values. This will lead to both a more concise swagger document, and (more importantly) much, much more useful client libraries. I'd be more than happy to sit down with you for 30min and walk through some of the changes we can make in the model definitions to better describe the service...